Sources/PackageGraph/PackageGraph.swift

```diff
@@ -33,8 +33,12 @@ public struct PackageGraph {
         return rootPackages.flatMap{$0.modules}.contains(module)
     }
 
+    /// The errors encountered while loading the package graph.
+    public let errors: [Swift.Error]
```

Comment:
Also, rather than collecting all the errors until the end (which could take a long time if big packages have to be cloned), might it perhaps be better to use delegation here so that things like error messages can be emitted sooner?